src/graph.ts

```diff
@@ -61,14 +62,20 @@ export function createDeepAgent<
     subagents = [],
     postModelHook,
     interruptConfig = {},
+    builtinTools,
   } = params;
 
   const stateSchema = params.stateSchema
     ? DeepAgentState.extend(params.stateSchema.shape)
     : DeepAgentState;
 
+  // Filter built-in tools if builtinTools parameter is provided
+  const selectedBuiltinTools = builtinTools
+    ? BUILTIN_TOOLS.filter((tool) => builtinTools.includes(tool.name))
```

Comment:
i prefer this incase any of the buildin tools have a substring of another builtin tool
```suggestion
    ? BUILTIN_TOOLS.filter((tool) => builtinTools.some((bt) => bt === tool.name))
```